main.go

```diff
@@ -126,26 +136,65 @@ func (c *graphiteCollector) processLines() {
 	}
 }
 
+func parseMetricNameAndTags(name string) (string, prometheus.Labels, error) {
+	var err error
+
+	labels := make(prometheus.Labels)
+
+	parts := strings.Split(name, ";")
+	parsedName := parts[0]
+
+	tags := parts[1:]
+	for _, tag := range tags {
+		kv := strings.SplitN(tag, "=", 2)
+		if len(kv) != 2 {
+			// don't add this tag, continue processing tags but return an error
+			tagParseFailures.Inc()
+			err = fmt.Errorf("error parsing tag %s", tag)
+			continue
+		}
+
+		k := kv[0]
+		v := kv[1]
+		labels[k] = v
+	}
+
+	return parsedName, labels, err
+}
+
 func (c *graphiteCollector) processLine(line string) {
 	line = strings.TrimSpace(line)
 	level.Debug(c.logger).Log("msg", "Incoming line", "line", line)
+
 	parts := strings.Split(line, " ")
 	if len(parts) != 3 {
 		level.Info(c.logger).Log("msg", "Invalid part count", "parts", len(parts), "line", line)
 		return
 	}
+
 	originalName := parts[0]
-	var name string
-	mapping, labels, present := c.mapper.GetMapping(originalName, mapper.MetricTypeGauge)
 
-	if (present && mapping.Action == mapper.ActionTypeDrop) || (!present && c.strictMatch) {
+	parsedName, labels, err := parseMetricNameAndTags(originalName)
+	if err != nil {
+		level.Info(c.logger).Log("msg", "Invalid tags", "line", line, "err", err.Error())
```

Comment:
Info feels like the wrong level here. It should either be a Warning since it's a problem, or Debug since it's user-input-driven and may be very high volume. To keep in line with the statsd exporter and the mapper package, I think it is the latter:
```suggestion
		level.Debug(c.logger).Log("msg", "Invalid tags", "line", line, "err", err.Error())
```